### PR TITLE
Add codelab: deploying Artifactory artifacts to Cloud Foundry

### PR DESCRIPTION
**Pull request by fifthposition**
This starts with an Artifactory repository (for a Maven artifact) and a GitHub repository (for the app manifest) and deploys a Maven artifact to CF.

### CODE REVIEW

**Comment by fifthposition** [comment]
cc @dorbin (reference info for the Artifactory trigger and Maven artifact etc. to come later)

File: guides/tutorials/codelabs/pubsub-to-cf/index.md [hunk 1]
```diff
@@ -62,10 +60,10 @@ b. Create a key for the service account and download the key in JSON format to y
 
 ## 4. Tell Spinnaker to Use the Pub/Sub Subscription
 
-a. Enable artifact support, then enable the GCS artifact provider:
+a. Enable the `artifactsRewrite` feature flag, then enable the GCS artifact provider:
```
Comment (dorbin) [inline guides/tutorials/codelabs/pubsub-to-cf/index.md]: @fifthposition Does flipping this flag affect any of the rest of the codelab?

File: guides/tutorials/codelabs/artifactory-to-cf/index.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+   nav: guides
+---
+
+ {% include toc %}
+ 
+> This codelab assumes that you have enabled the `artifactsRewrite` feature flag. In `~/.hal/$DEPLOYMENT/profiles/settings-local.js` (where `$DEPLOYMENT` is typically `default`), add:
+>
+> `window.spinnakerSettings.feature.artifactsRewrite = true;`
+
+In this codelab, you will deploy an artifact to Cloud Foundry via a Spinnaker pipeline that is triggered by an artifact published to a JFrog Artifactory Maven repository or by an app manifest stored in a GitHub Git repository.
```
Comment (dorbin) [inline guides/tutorials/codelabs/artifactory-to-cf/index.md]:
```suggestion
In this codelab, you will deploy an artifact to Cloud Foundry via a Spinnaker pipeline. The pipeline is triggered by an artifact published to a JFrog Artifactory Maven repository or by an app manifest stored in a GitHub repository.
```

File: guides/tutorials/codelabs/artifactory-to-cf/index.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+## Prerequisites
+
+This codelab assumes you have the following:
+
+1. An Artifactory repository configured to accept a Maven artifact--the JAR for your application.
```
Comment (dorbin) [inline guides/tutorials/codelabs/artifactory-to-cf/index.md]:
```suggestion
* An Artifactory repository configured to accept a Maven artifact&mdash;the JAR for your application
```

File: guides/tutorials/codelabs/artifactory-to-cf/index.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+1. A GitHub repository containing a manifest with which to deploy your application.
```
Comment (dorbin) [inline guides/tutorials/codelabs/artifactory-to-cf/index.md]:
```suggestion
* A GitHub repository containing a manifest with which to deploy your application
```

File: guides/tutorials/codelabs/artifactory-to-cf/index.md [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+     image_path="./git-trigger.png"
+  %}
+
+d. In the Artifact Constraints dropdown for the Git trigger, select "Define a new artifact..." and populate the Expected Artifact form with the details of the manifest stored in the GitHub repository.
+
+  {% include figure
+     image_path="./git-expected-artifact.png"
+  %}
+
+e. Add a "Deploy" stage to the pipeline. This stage will deploy to the test environment.
```
Comment (dorbin) [inline guides/tutorials/codelabs/artifactory-to-cf/index.md]:
```suggestion
e. Add a "Deploy" stage to the pipeline. This stage deploys to the test environment.
```

File: guides/tutorials/codelabs/artifactory-to-cf/index.md [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+     image_path="./pipeline-second-stage.png"
+  %}
+
+Add any desired configuration, such as notifications for when the stage is awaiting judgment or is manually judged to continue or stop.
+
+  {% include figure
+     image_path="./manual-judgment-configuration.png"
+  %}
+
+After you have validated and approved the deployment, the next stage (following this "Manual Judgment" stage) will deploy to the production environment.
```
Comment (dorbin) [inline guides/tutorials/codelabs/artifactory-to-cf/index.md]:
```suggestion
After you have validated and approved the deployment, the next stage (following this "Manual Judgment" stage) deploys to the production environment.
```

File: guides/tutorials/codelabs/artifactory-to-cf/index.md [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+
+h. Add a "Clone Server Group" stage to the pipeline. This stage will promote the deployed application to the production environment.
```
Comment (dorbin) [inline guides/tutorials/codelabs/artifactory-to-cf/index.md]:
```suggestion
h. Add a "Clone Server Group" stage to the pipeline. This stage promotes the deployed application to the production environment.
```

File: guides/tutorials/codelabs/artifactory-to-cf/index.md [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+     image_path="./pipeline-third-stage.png"
+  %}
+
+i. Add a clone configuration and provide details on deployment settings, the source server group, and the manifest:
+
+  {% include figure
+     image_path="./clone-configuration.png"
+  %}
+
+This clone configuration will promote the exact same application to the production environment as you deployed to the test environment with the "Deploy" stage in Step 5e. (You may wish to provide a different manifest when deploying the application to the production environment. For this alternate manifest to trigger pipeline executions, you would need to create a separate pipeline that deploys to the production environment.)
```
Comment (dorbin) [inline guides/tutorials/codelabs/artifactory-to-cf/index.md]:
```suggestion
This clone configuration promotes the exact same application to the production environment as you deployed to the test environment with the "Deploy" stage in Step 5e. 

You may wish to provide a different manifest when deploying the application to the production environment. For this alternate manifest to trigger pipeline executions, you would need to create a separate pipeline that deploys to the production environment.
```

File: guides/tutorials/codelabs/artifactory-to-cf/index.md [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+
+## 6. Publish the Application Artifact or Commit Configuration Changes
+
+a. Publish the application artifact to your Artifactory repository. You can also update the application manifest and push a new commit to its GitHub repository.
+
+b. The published artifact or Git repository push will trigger a pipeline execution, and you should see the pipeline deploy a new server group for the application, then promote the application to staging.
```
Comment (dorbin) [inline guides/tutorials/codelabs/artifactory-to-cf/index.md]:
```suggestion
b. The published artifact or Git repository push triggers your pipeline, and you should see it deploy a new server group for the application, then promote the application to staging.
```

**Review by dorbin — COMMENTED** [review]
Just a few editorial suggestions. 
Also, you'll want to add an entry to `/_data/navigation.yml`

File: guides/tutorials/codelabs/pubsub-to-cf/index.md (hunk 1 above)
Comment (fifthposition) [reply guides/tutorials/codelabs/pubsub-to-cf/index.md]: Oh dear... looks like a change snuck in that I hadn't intended to commit. I'll revert this one. Sorry!

**Review by dorbin — APPROVED** [review]
:+1: 
Thanks, @fifthposition

**Pull request merged**